mcs/class/Mono.Debugger.Soft/Mono.Debugger.Soft/Connection.cs

```diff
@@ -845,9 +850,12 @@ class PacketReader {
 					return new ValueImpl { Type = etype, Value = ReadDouble () };
 				case ElementType.I:
 				case ElementType.U:
-				case ElementType.Ptr:
 					// FIXME: The client and the debuggee might have different word sizes
 					return new ValueImpl { Type = etype, Value = ReadLong () };
+				case ElementType.Ptr:
+					long value = ReadLong();
+					long pointerClass = ReadId();
+					return new ValueImpl {Type = etype, Klass = pointerClass, Value = value};
```

Comment:
This is a breaking protocol change, so it needs to be inside a:
								if (Version.AtLeast (2, 46))

Reply:
@vargaz this is inside the PacketReader class that doesn't have access to connection/vm to check version. What is best way to handle this?

Reply:
Pass the connection instance to the PacketReader ctor.